### ISSUE TITLE
BA on level 1 of a building unable to move to adjacent building

### ISSUE THREAD

**Issue by Question2** [MegaMek/megamek#204]
The first BA squad in this savegame cannot move to any adjacent building hexes even though it is on floor 1 of the building hex. The second BA squad in the same hex can move to adjacent building hexes, and after moving the second BA squad, the first BA squad can walk/run into adjacent building hexes normally.

[#battletech.zip](https://github.com/MegaMek/megamek/files/271875/battletech.zip)

**Comment by BLOODWOLF333** [comment]
Loaded saved game and could not duplicate concern.  EBA [MG2}#1-4 all were allowed to move to adjacent buildings.  I moved #1 (blank) first and it did move.  Is that the specific EBA you were talking about?

**Comment by Question2** [comment]
Yes. I tried to load the save game but could not duplicate the issue either, I'm guessing the problem resolves itself if you close MM and load the save game again.

**Comment by Question2** [comment]
I think I found the cause.

With the BA selected, click jump, then press ESC to reset back to walk mode. I wasn't able to move into the adjacent building afterwards.

![Uploading megamek.jpg…]()

**Comment by arlith** [comment]
Pressing escape doesn't reset the movement mode anymore, it just clears the move path.  You're still in "jump" mode, if you click Jump and then press ESC.  It's more clear if you've got the "movement envelope" turned on.  I wasn't able to reproduce the issue with the attached saved game.  I also didn't see any situations like the one depicted in your screenshot.  Do you have a different saved game?  It does look like you've got Walk mode on in that screenshot (teal arrows, vs red arrows).

**Comment by Question2** [comment]
Okay thats a weird change then, sicne it makes it much less convenient to reset movement from jump to walk.

I tested it again with 2 standard IS BA on an urban map. This is what I noticed : 

-If i click jump, then click on a hex, then click walk, the next time i click on a hex, i will still be in jump mode, not walk mode. I have to click walk AGAIN to get into jump mode.

-After clicking jump once, the BA becomes bugged and will not walk into some hexes. In this case, it was the NW, N and NE hexes.

-The BA can jump into an adjacent buiding hex...using just one jump MP, it always ends up on level 3 of the building, even though it started at level 1. It doesn't need to make an anti-mek skill roll to enter the building via jump MP either (page 168, TW).

Player name is #battletech.

[test.sav.gz](https://github.com/MegaMek/megamek/files/278853/test.sav.gz)

![megamek](https://cloud.githubusercontent.com/assets/1191378/15493518/dfc4b1c4-21c7-11e6-85a9-8279c8a1136b.jpg)

**Comment by arlith** [comment]
There was a bug related to switching from jump to walk mode, which is fixed in e5f701c.  I think that is ultimately what is causing the issue here.  I'm going to close this ticket for now, but you can comment if you still have the issue in 41.22 or later.

**Comment by Question2** [comment]
I am having a very similar issue (may be the same one) in 41.25.

![megamek](https://cloud.githubusercontent.com/assets/1191378/19624647/af2cda58-994c-11e6-9dc5-c74a30c5ceae.jpg)

1.Select the first elemental squad in the list. Go into jump mode, click somewhere to get the red jump numbers to show up, then switch bak to walk mode with the walk mode

2. Try to walk into the adjacent building as shown in the screenshot. Megamek will calculate the MP cost incorrectly and will refuse to let you walk into the building.

If you restart megamek, the bug will resolve itself and you can walk into the buildings normally, until you go into jump mode again.

[#battletech.sav.gz](https://github.com/MegaMek/megamek/files/546225/battletech.sav.gz)
[mekhqlog.txt](https://github.com/MegaMek/megamek/files/546226/mekhqlog.txt)

**Comment by arlith** [comment]
Ok, with the latest report, I've been able to reproduce the issue and figure out what the problem is.  What's happening is there is an `Entity.isJumpingNow` state which was added in bdc5e47d.  This gets used in `Entity.calcElevation` to determine if a unit should be placed on top of a building, or should enter at the current elevation.  This state variable gets set when a `MovePath` is compiled for a jump, and when a new round begins.  The problem was, if a jump movement is planned, the `MovePath` gets compiled, and that state gets set, however it doesn't then get cleared until the next round.  To fix this, I made `MovementDisplay.clear()` set the `Entity.isJumpingNow` state to false.
